src/tasks/task-se3-equality.cpp

```diff
@@ -196,8 +196,8 @@ namespace tsid
       m_v_ref_vec = m_v_ref.toVector();
       m_v = v_frame.toVector();
 
-      m_constraint.setMatrix(m_mask.transpose() * m_J);
-      m_constraint.setVector(m_mask.transpose() * (m_a_des - m_drift.toVector()));
+      m_constraint.setMatrix(m_mask.asDiagonal() * m_J);
+      m_constraint.setVector(m_mask.asDiagonal() * (m_a_des - m_drift.toVector()));
```

Comment:
In line 200 it may be more efficient to use `cwiseProduct` rather than `asDiagonal`